### PR TITLE
feat(0x): added pool trade via 0x

### PR DESCRIPTION
**Pull request by kent426**

### CODE REVIEW

**Review by hochbruj — APPROVED** [review]
Great job! I got test cases on both networks optimism and polygon successfully finished.

File: src/config.ts [hunk 1]
```diff
@@ -93,6 +96,12 @@ export const networkChainIdMap: NetworkChainIdMap = {
   [Network.ARBITRUM]: 42161
 };
 
+export const zeroExBaseUrlMap: ZeroExBaseUrlMap = {
+  [Network.POLYGON]: "https://polygon.api.0x.org",
+  [Network.OPTIMISM]: "https://optimism.api.0x.org",
+  [Network.ARBITRUM]: "https://arbitrum.api.0x.org"
+};
+
```
[the remark this answers is not in the document]
Comment (kent426) [reply src/config.ts]: make sense

File: src/services/zeroEx/zeroExTrade.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      // necessary to skip quote validation is that in which the takerAddress refers to a smart contract
+      skipValidation: true,
+      // Used to enable RFQ-T liquidity
+      intentOnFilling: true,
+      takerAddress,
+      slippagePercentage
+      // excludedSourcesParam
+    };
+    const response = await axios.get(
+      `${zeroExBaseUrlMap[network]}/swap/v1/quote`,
```
[the remark this answers is not in the document]
Comment (kent426) [reply src/services/zeroEx/zeroExTrade.ts]: just pushed an update to resolve it

File: src/test/zeroEx.test.ts
[the remark this answers is not in the document]
Comment (kent426) [reply src/test/zeroEx.test.ts:1]: Yes, I think I miss out how to do local testing properly, so I did tests on-chain with specific dhedge pools. If local testing is sufficient, then we can delete the on-chain testing and keep it consistent.

**Pull request merged**